lib/HtmlPanel.tsx

```diff
@@ -0,0 +1,94 @@
+import { useContext, useEffect, useState } from "react";
+import { createPortal } from "react-dom";
+import Talk from "talkjs";
+import { BoxContext } from "./MountedBox";
+
+type HtmlPanelProps = {
+  /**
+   * The URL you want to load inside the HTML panel. The URL can be absolute or
+   * relative. We recommend using same origin pages to have better control of
+   * the page. Learn more about HTML Panels and same origin pages {@link https://talkjs.com/docs/Features/Customizations/HTML_Panels/ | here}.
+   */
+  url: string;
+
+  /** The panel height in pixels. Defaults to `100px`. */
+  height?: number;
+
+  /** Sets the visibility of the panel. Defaults to `true`. */
+  show?: boolean;
+
+  /** If given, the panel will only show up for the conversation that has an `id` matching the one given. */
+  conversationId?: string;
+
+  /** The content that gets rendered inside the `<body>` of the panel. */
+  children: React.ReactNode;
+};
+
+type State =
+  | { type: "none" }
+  | { type: "loading" }
+  | { type: "loaded"; panel: Talk.HtmlPanel };
+
+export function HtmlPanel({
+  url,
+  height = 100,
+  show = true,
+  conversationId,
+  children,
+}: HtmlPanelProps) {
+  const [state, setState] = useState<State>({ type: "none" });
+  const box = useContext(BoxContext);
+
+  useEffect(() => {
+    async function run() {
+      if (state.type !== "none" || !box) return;
+
+      setState({ type: "loading" });
+      const panel = await box.createHtmlPanel({
+        url,
+        conversation: conversationId,
+        height,
+        show,
+      });
```

Comment:
meta comment, I just realized that a set of `<HtmlPanel>` components in a react app is semantically equivalent to the `htmlPanelOptions` we pass from the JSSDK in the frontend.

this is actually super nuts. we internally have a declarative data structure, we built an imperative interface around that to mutate that data structure (createHtmlPanel, destroy, show, etc), and now here we're building the same thing in the opposite direction!

i don't think we need to do anything rn with this observation but i'm thinking that maybe one day we should deprecate all `create*` and `set*` methods entirely in favour of some vanilla JS, react-y declarative "spec your chatbox here" interface. would make the entire react sdk a near no-op, also.